src/ast/inline-document-descriptor.ts

```diff
@@ -57,33 +57,39 @@ export class InlineParsedDocument<N> implements ScannedFeature {
 
   scannedDocument: ScannedDocument;
 
+  sourceRange: SourceRange;
+
   constructor(
       type: string, contents: string, node: N, locationOffset: LocationOffset,
-      attachedComment: string) {
+      attachedComment: string, sourceRange: SourceRange) {
     this.type = type;
     this.contents = contents;
     this.node = node;
     this.locationOffset = locationOffset;
     this.attachedComment = attachedComment;
+    this.sourceRange = sourceRange;
   }
 }
 
-export function correctSourceLocation(
-    sourceLocation: SourceLocation,
-    locationOffset?: LocationOffset): SourceLocation|undefined {
-  if (!locationOffset || !sourceLocation) {
-    return sourceLocation;
+export function correctSourceRange(
+    sourceRange: SourceRange, locationOffset?: LocationOffset): SourceRange|
+    undefined {
```

Comment:
That only works on identifier names, not on type names, so AIUI this is the only syntax for annotating return values.